src/classes/WordPress.php

```diff
@@ -44,7 +46,13 @@ public function __construct() {
 			add_filter( 'file_mod_allowed', array( $this, 'read_only_filesystem' ), PHP_INT_MAX, 2 );
 			add_filter( 'pre_reschedule_event', array( $this, 'delay_cronjobs' ), PHP_INT_MAX, 2 );
 			add_action( 'admin_enqueue_scripts', array( $this, 'admin_scripts' ) );
-			add_action( 'admin_init', array( &$this, 'check_block_request' ) );
+			add_action( 'admin_init', array( $this, 'check_block_request' ) );
+
+			// Runs only when WP_REWRITE_URLS is set to true and for staging stores
+			if ( $this->is_rewrite_urls() && $this->is_staging() ) {
+				add_action( 'wp_loaded', array( $this, 'start_buffering' ), ~PHP_INT_MAX );
+				add_action( 'wp_print_footer_scripts', array( $this, 'end_buffering' ), PHP_INT_MAX );
```

Comment:
What about https://www.php.net/manual/en/function.register-shutdown-function.php ?

Reply:
Yup, we can use that too.

In fact, now I'm thinking about it, `register_shutdown_function` will cover edge cases as well where any theme adds anything below `wp_footer` in the template.

Thanks for suggesting it :)